### PR TITLE
feat: Make `pint deploy` package-aware. For pint packages, build the package before deploying it.

### PR DESCRIPTION
**Pull request by mitchmindtree**
## To-Do

- [x] Land and publish changes in https://github.com/essential-contributions/pint/pull/995.
- [x] Update toml to point to latest `pint-pkg`, `pint-cli`.

Closes https://github.com/essential-contributions/essential-integration/issues/123

## Follow-up

- https://github.com/essential-contributions/essential-integration/issues/129

### CODE REVIEW

**Review by mohammadfawaz — APPROVED** [review]

File: crates/pint-deploy/src/main.rs [hunk 1]
```diff
@@ -1,43 +1,129 @@
...
+use std::path::{Path, PathBuf};
 
 #[derive(Parser, Debug)]
-#[command(version, about, long_about = None)]
+#[command(name = "deploy", version, about, long_about = None)]
+#[group(skip)]
 /// Tool to deploy a contract to a Essential builder endpoint.
 struct Args {
-    /// The endpoint of builder to bind to.
+    // `pint deploy` builds too, so accepts all args that
```
Comment (nfurfaro) [inline crates/pint-deploy/src/main.rs:13]:
```suggestion
    // `pint deploy` builds too, so accepts all args that `pint build` does.
```

File: crates/pint-deploy/src/main.rs [hunk 2]
```diff
@@ -1,43 +1,129 @@
...
+    #[command(flatten)]
+    build_args: pint_cli::build::Args,
+    /// The builder to which the contract deployment solution will be submitted.
     #[arg(long)]
     builder_address: String,
-    /// Path to the contract file as a json `Contract`.
+    /// Path to a specific `Contract` encoded as JSON.
+    ///
+    /// If a specific contract is specified in this manner, all arguments related to building a
```
Comment (nfurfaro) [inline crates/pint-deploy/src/main.rs:21]:
```suggestion
    /// If a contract is specified in this manner, all arguments related to building a
```

**Review by nfurfaro — APPROVED** [review]
Just a couple tiny nits. Nice work!

**Review by supiket — APPROVED** [review]
So neat with the details, especially with the command outputs 🤯

**Comment by supiket** [comment]
Closes #123, right?

**Review by nfurfaro — APPROVED** [review]

**Pull request merged**